gb-cpu/src/memory/area.rs

```diff
@@ -6,28 +6,28 @@ pub use wram::Wram;
 
 #[derive(Debug, PartialEq, Eq)]
 pub enum Area {
-    Rom,
-    Vram,
-    ExtRam,
-    Wram,
-    EchoRam,
-    Oam,
-    IOReg,
-    HighRam,
+    Rom(u16),
+    Vram(u16),
+    ExtRam(u16),
+    Wram(u16),
+    EchoRam(u16),
+    Oam(u16),
+    IOReg(u16),
+    HighRam(u16),
     IEReg,
 }
 
 impl Area {
-    pub fn relative(&self, address: u16) -> usize {
+    pub fn relative(&self) -> usize {
         let result = match self {
-            Area::Rom => address,
-            Area::Vram => address - consts::ROM_MIN,
-            Area::ExtRam => address - consts::EXT_RAM_MIN,
-            Area::Wram => address - consts::WRAM_MIN,
-            Area::EchoRam => address - consts::ECHO_RAM_MIN,
-            Area::Oam => address - consts::OAM_MIN,
-            Area::IOReg => address - consts::IOREG_MIN,
-            Area::HighRam => address - consts::HIGH_MIN,
+            Area::Rom(address) => *address,
+            Area::Vram(address) => *address - consts::ROM_MIN,
+            Area::ExtRam(address) => *address - consts::EXT_RAM_MIN,
+            Area::Wram(address) => *address - consts::WRAM_MIN,
+            Area::EchoRam(address) => *address - consts::ECHO_RAM_MIN,
+            Area::Oam(address) => *address - consts::OAM_MIN,
+            Area::IOReg(address) => *address - consts::IOREG_MIN,
+            Area::HighRam(address) => *address - consts::HIGH_MIN,
             Area::IEReg => consts::INTERUPT_ENABLE,
```

Comment:
The relative position in the `IEReg` should be 0 ?

Reply:
Yeah that's an excess of code, great point!